docs/visual-basic/programming-guide/concepts/linq/basic-query-operations.md

```diff
@@ -16,140 +16,134 @@ helpviewer_keywords:
- You can combine more than one data source in the `From` clause in several ways. For example, the following code uses two data sources and implicitly combines properties from both of them in the result. The query selects students whose last names start with a vowel.  
-  
- [!code-vb[VbLINQBasicOps#9](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbLINQBasicOps/VB/Class1.vb#9)]  
+# Basic query operations (Visual Basic)
+
+This topic provides a brief introduction to [!INCLUDE[vbteclinqext](~/includes/vbteclinqext-md.md)] expressions in Visual Basic, and to some of the typical kinds of operations that you perform in a query. For more information, see the following topics:
+
+- [Introduction to LINQ in Visual Basic](../../language-features/linq/introduction-to-linq.md)
+- [Queries](../../../language-reference/queries/index.md)
+- [Walkthrough: Writing Queries in Visual Basic](walkthrough-writing-queries.md)
+
+## Specify the data source (`From`)
+
+In a [!INCLUDE[vbteclinq](~/includes/vbteclinq-md.md)] query, the first step is to specify the data source that you want to query. Therefore, the `From` clause in a query always comes first. Query operators select and shape the result based on the type of the source.
+
+[!code-vb[VbLINQBasicOps#1](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbLINQBasicOps/VB/Class1.vb#1)]
+
+The `From` clause specifies the data source, `customers`, and a *range variable*, `cust`. The range variable is like a loop iteration variable, except that in a query expression, no actual iteration occurs. When the query is executed, often by using a `For Each` loop, the range variable serves as a reference to each successive element in `customers`. Because the compiler can infer the type of `cust`, you do not have to specify it explicitly. For examples of queries written with and without explicit typing, see [Type Relationships in Query Operations (Visual Basic)](type-relationships-in-query-operations.md).
+
+For more information about how to use the `From` clause in Visual Basic, see [From Clause](../../../language-reference/queries/from-clause.md).
+
+## Filter data (`Where`)
+
+Probably the most common query operation is applying a filter in the form of a Boolean expression. The query then returns only those elements for which the expression is true. A `Where` clause is used to perform the filtering. The filter specifies which elements in the data source to include in the resulting sequence. In the following example, only those customers who have an address in London are included.
+
+[!code-vb[VbLINQBasicOps#2](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbLINQBasicOps/VB/Class1.vb#2)]
+
+You can use logical operators such as `And` and `Or` to combine filter expressions in a `Where` clause. For example, to return only those customers who are from :::no-loc text="London"::: and whose name is :::no-loc text="Devon":::, use the following code:
+
+```vb
+Where cust.City = "London" AndAlso cust.Name = "Devon"
```

Comment:
**First reason: (most important one)** is short circuiting. If the first condition is False. The second won't be evaluated when AndAlso is used, but it will be evaluated (which isn't redundant) when And is used.

**Second reason:** To avoid confusion. Someone would say, why Bitwise operation is performed while the operation needed is logical operation. For boolean, both have same output for sure, but I think it's better to emphasize the fact that it's logical not bitwise.

**Third reason: ** This may be a personal opinion, but I think it's better practice to use logical operator in this case.